packages/react-components/react-datepicker-compat/src/components/Calendar/Calendar.tsx

```diff
@@ -297,7 +297,10 @@ export const Calendar: React.FunctionComponent<CalendarProps> = React.forwardRef
           break;
       }
     };
-    const showMonthPickerAsOverlay = getShowMonthPickerAsOverlay(props);
+    const showMonthPickerAsOverlay = getShowMonthPickerAsOverlay({
+      isDayPickerVisible: isDayPickerVisibleProp,
+      showMonthPickerAsOverlay: showMonthPickerAsOverlayProp,
+    });
```

Comment:
tldr, we need to use the extracted props since passing the props object doesn't account for default values.